src/main/scala/com/exasol/cloudetl/bucket/Bucket.scala

```diff
@@ -54,7 +55,12 @@ abstract class Bucket extends LazyLogging {
     getRequiredProperties().foreach { key =>
       if (!properties.containsKey(key)) {
         throw new IllegalArgumentException(
-          s"Please provide a value for the $key property!"
+          ExaError
+            .messageBuilder("E-CSE-2")
+            .message("Required {{KEY}} property value is missing.")
+            .parameter("KEY", key)
```

Comment:
you can also do:
```suggestion
            .message("Required {{KEY}} property value is missing.", key)
```

Reply:
This is nice, definitely more nicer for some of them.